purchase_order_secondary_unit/views/purchase_order_views.xml

```diff
@@ -11,42 +11,42 @@
                 expr="//field[@name='order_line']/form//field[@name='product_qty']"
                 position="before"
             >
-                <t groups="uom.group_uom">
-                    <field
-                        name="secondary_uom_qty"
-                        class="oe_inline oe_no_button"
-                        attrs="{'readonly': [('state', 'in', ('done', 'cancel'))]}"
-                    />
-                    <field
-                        name="secondary_uom_id"
-                        class="oe_inline"
-                        domain="product_id and ['|', ('product_id', '=', product_id),
-                                    '&amp;', ('product_tmpl_id.product_variant_ids', 'in', [product_id]),
-                                            ('product_id', '=', False)] or [(0, '=', 1)]"
-                        options="{'no_create': True}"
-                        attrs="{'readonly': [('state', 'in', ('purchase', 'done', 'cancel'))], 'required': [('secondary_uom_qty', '!=', 0.0)]}"
-                    />
-                </t>
+                <field
+                    name="secondary_uom_qty"
+                    class="oe_inline oe_no_button"
+                    groups="uom.group_uom"
+                    readonly="state in ['done', 'cancel']"
+                />
+                <field
+                    name="secondary_uom_id"
+                    class="oe_inline"
+                    domain="['|', ('product_id', '=', product_id),
+                                '&amp;', ('product_tmpl_id.product_variant_ids', 'in', [product_id]),
+                                         ('product_id', '=', False)]"
+                    options="{'no_create': True}"
+                    groups="uom.group_uom"
+                    readonly="state in ['purchase','done', 'cancel']"
+                    required="secondary_uom_qty != 0.0"
+                />
             </xpath>
             <xpath
                 expr="//field[@name='order_line']/tree//field[@name='product_qty']"
                 position="before"
             >
                 <field
                     name="secondary_uom_qty"
-                    attrs="{'readonly': [('state', 'in', ('done', 'cancel'))]}"
                     groups="uom.group_uom"
-                    optional="show"
```

Comment:
Please don't remove this, as it was added in an recent commit, when updating commit history